iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/thread.hpp

```diff
@@ -28,8 +33,64 @@ constexpr uint64_t MAX_THREAD_NAME_LENGTH = 15U;
 
 using ThreadName_t = cxx::string<MAX_THREAD_NAME_LENGTH>;
 
-void setThreadName(pthread_t thread, const ThreadName_t& name) noexcept;
-ThreadName_t getThreadName(pthread_t thread) noexcept;
+/// @todo remove free functions
+void setThreadName(iox_pthread_t thread, const ThreadName_t& name) noexcept;
+ThreadName_t getThreadName(iox_pthread_t thread) noexcept;
+
+enum class ThreadError
+{
+    EMPTY_CALLABLE,
+    INSUFFICIENT_MEMORY,
+    INSUFFICIENT_PERMISSIONS,
+    INSUFFICIENT_RESOURCES,
+    INVALID_ATTRIBUTES,
+    UNDEFINED
+};
+
+class Thread
```

Comment:
Could you add a doxygen documentation for the class and add a code snippet explaining how to use it.